src/pages/experiments/[experimentId]/plots-and-tables/embedding-categorical/index.jsx

```diff
@@ -116,33 +154,22 @@ const EmbeddingCategoricalPlot = () => {
         />
       );
     }
-    if (!config || !data || loading || !isBrowser) {
+
+    if (!config || !data || loading) {
       return (
         <center>
           <Spin size='large' />
         </center>
       );
     }
+
     return (
       <center>
-        <Vega spec={vegaSpec} renderer='canvas' />
+        <Vega spec={plotSpec} renderer='canvas' />
       </center>
     );
   };
 
-  const vegaSpec = generateSpec(config);
-  // due to a bug in vega with React with using data with source coming from other data,
-  // we have to inject the data in the Vega spec.
-  generateData(vegaSpec);
```

Comment:
Doing stuff like this is kind of bad, because it will not allow us to load the link directly from the browser, which I found out when I was looking for the bug.